cfme/automate/explorer/domain.py

```diff
@@ -82,200 +84,27 @@ def is_displayed(self):
-        else:
-            # Not changed
-            priority_page.cancel_button.click()
-        domains_view = self.create_view(DomainListView)
-        assert domains_view.is_displayed
-        domains_view.flash.assert_no_error()
-        if changed:
-            domains_view.flash.assert_message('Priority Order was saved')
-        else:
-            domains_view.flash.assert_message('Edit of Priority Order was cancelled by the user')
-        return changed
-
-
-@navigator.register(DomainCollection)
-class All(CFMENavigateStep):
-    VIEW = DomainListView
-    prerequisite = NavigateToAttribute('appliance.server', 'AutomateExplorer')
-
-    def step(self):
-        self.prerequisite_view.datastore.tree.click_path(*self.obj.tree_path)
-
-
-@navigator.register(DomainCollection)
-class Add(CFMENavigateStep):
-    VIEW = DomainAddView
-    prerequisite = NavigateToSibling('All')
-
-    def step(self):
-        self.prerequisite_view.configuration.item_select('Add a New Domain')
-
-
-@navigator.register(DomainCollection)
-class Priority(CFMENavigateStep):
-    VIEW = DomainPriorityView
-    prerequisite = NavigateToSibling('All')
-
-    def step(self):
-        self.prerequisite_view.configuration.item_select('Edit Priority Order of Domains')
-
-
-class DomainDetailsView(AutomateExplorerView):
-    title = Text('#explorer_title_text')
-    namespaces = Table('#ns_details_grid')
-
-    @property
-    def is_displayed(self):
-        return (
-            self.in_explorer and
-            self.title.text == 'Automate Domain "{}"'.format(
-                self.context['object'].table_display_name))
-
-
+@attr.s
 class Domain(BaseEntity, Fillable):
     """A class representing one Domain in the UI."""
-    def __init__(
-            self, collection, name, description, enabled=None, locked=None,
-            git_repository=None, git_checkout_type=None, git_checkout_value=None, db_id=None):
-        if db_id is not None:
-            self.db_id = db_id
-        self.collection = collection
-        self.appliance = self.collection.appliance
-        self.name = name
-        self.description = description
-        if git_repository is not None:
-            self.git_repository = git_repository
-        if git_checkout_type is not None:
-            self.git_checkout_type = git_checkout_type
-        if git_checkout_value is not None:
-            self.git_checkout_value = git_checkout_value
-        if enabled is not None:
-            self.enabled = enabled
-        if locked is not None:
-            self.locked = locked
+    name = attr.ib()
+    description = attr.ib()
+    enabled = attr.ib(default=None)
+    locked = attr.ib(default=None)
+    git_repository = attr.ib(default=None)
+    git_checkout_type = attr.ib(default=None)
+    git_checkout_value = attr.ib(default=None)
+    db_id = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        fn = lambda *a, **k: Domain._db_id(self, *a, **k)
+        self.db_id = self.db_id or fn
```

Comment:
this one looks like a structural issue, we should either use a factory function that takes self, or defer it altogether

based on the similar patterns below we might need to consider ways for lazyly obtaining such attributes
the current solution looks workable thus we just need a task to tackle this later on

Reply:
Oh this one plain doesn't work